starky/src/lookup.rs

```diff
@@ -577,7 +580,7 @@ pub(crate) fn lookup_helper_columns<F: Field>(
     constraint_degree: usize,
 ) -> Vec<PolynomialValues<F>> {
     assert!(
-        constraint_degree == 2 || constraint_degree == 3,
+        constraint_degree <= 3,
         "TODO: Allow other constraint degrees."
```

Comment:
Actually this was backported from the evm side but I'm not sure I understand the restriction. As long as `constraint_degree <= blowup_factor + 1`, this should be fine, no?

Reply:
Maybe because <= 3 was a requirement when this code was in the zk_evm?

Reply:
Agree, given that there are no zk_evms here.

Reply:
The only problem is that here we don't know the blowup_factor. But why is this assert here in the first place?

Reply:
I think it's just to not go through an expensive proof computation if the parameters were badly set. We can get rid of it, or adapt it to take into consideration the blowup factor.

Reply:
But one of the two decisions needs to be taken before this is merged.

Reply:
> I think it's just to not go through an expensive proof computation

Indeed. I mean, why is the assertion there if the file is about lookups. That's why I moved the assertion to `prove_with_commitment`, where you also happen to have access to the `rate_bits`.